lib/seahorn/BvOpSem2.cc

```diff
@@ -835,13 +835,128 @@ class OpSemVisitor : public InstVisitor<OpSemVisitor>, OpSemVisitorBase {
         m_ctx.setInstruction(*me);
       }
     } break;
+    case Intrinsic::sadd_with_overflow: {
+      Type *ty = I.getOperand(0)->getType();
+      Expr op0 = lookup(*I.getOperand(0));
+      Expr op1 = lookup(*I.getOperand(1));
+
+      // assume doAdd has same bitwidth as Op0
+      Expr add_res = m_ctx.alu().doAdd(op0, op1, ty->getScalarSizeInBits());
+      Expr is_overflow =
+          m_ctx.alu().IsSaddNoOverflow(op0, op1, ty->getScalarSizeInBits());
+      Expr is_underflow =
+          m_ctx.alu().IsBaddNoUnderflow(op0, op1, ty->getScalarSizeInBits());
+      Expr is_carry = m_ctx.alu().doNot(
+          m_ctx.alu().doAnd(is_overflow, is_underflow, 1 /* bitwidth */),
+          1 /* bitwidth*/);
+      Expr carry_1bv = m_ctx.alu().boolToBv1(is_carry);
```

Comment:
would be nice to hide `boolToBv1` inside `ALU` since this is a decision internal to the ALU

(by 'this', I mean the decision to represent `bv1` by Boolean instead of bit-vectors)